R/best_case_worst_case_transition_risk_profile_at_company_level.R

```diff
@@ -0,0 +1,88 @@
+best_case_worst_case_transition_risk_profile_at_company_level <- function(data) {
+  product <- data |>
+    unnest_product()
+
+  avg_best_case_worst_case_at_product_level <- product |>
+    create_avg_best_case_worst_case_at_product_level() |>
+    prepare_for_join_at_company_level()
+
+  company <- data |>
+    unnest_company() |>
+    left_join(avg_best_case_worst_case_at_product_level, by = c(col_companies_id(),
+      "benchmark_tr_score_avg" = col_transition_risk_grouped_by()
+    )) |>
+    polish_best_case_worst_case_transition_risk_at_company_level()
+
+  tilt_profile(nest_levels(product, company))
+}
+
+create_avg_best_case_worst_case_at_product_level <- function(data) {
+  crucial_cols <- c(
+    col_companies_id(), col_europages_product(),
+    col_transition_risk_grouped_by(), col_transition_risk_category(),
+    "amount_of_distinct_products"
+  )
+  check_crucial_cols(data, crucial_cols)
+
+  data |>
+    select(crucial_cols) |>
+    count_min_risk_category_per_company_benchmark() |>
+    count_max_risk_category_per_company_benchmark() |>
+    add_avg_transition_risk_best_case() |>
+    add_avg_transition_risk_worst_case()
+}
+
+count_min_risk_category_per_company_benchmark <- function(data) {
```

Comment:
Maybe DRY by passing "min" or "max" as an argument?

``` r
add_extreeme <- function(data, col = c("min", "max")) {
  # https://design.tidyverse.org/enumerate-options.html#whats-the-pattern
  col <- rlang::arg_match(col)
  
  # https://rlang.r-lib.org/reference/glue-operators.html
  dplyr::mutate(data, "n_{col}" := 1)
}

data <- data.frame(x = 1:3)

add_extreeme(data, "min")
#>   x n_min
#> 1 1     1
#> 2 2     1
#> 3 3     1

add_extreeme(data, "max")
#>   x n_max
#> 1 1     1
#> 2 2     1
#> 3 3     1

# Errors
add_extreeme(data, "bad")
#> Error in `add_extreeme()`:
#> ! `col` must be one of "min" or "max", not "bad".
```

<sup>Created on 2024-06-18 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>

Reply:
Thanks for the suggestion Mauro :)